src/ddmd/console.d

```diff
@@ -8,116 +8,206 @@
+    green     = 2,
+    blue      = 4,
+    yellow    = red | green,
+    magenta   = red | blue,
+    cyan      = green | blue,
+    white     = red | green | blue,
 }
 
-alias COLOR_BLACK = COLOR.COLOR_BLACK;
-alias COLOR_RED = COLOR.COLOR_RED;
-alias COLOR_GREEN = COLOR.COLOR_GREEN;
-alias COLOR_BLUE = COLOR.COLOR_BLUE;
-alias COLOR_YELLOW = COLOR.COLOR_YELLOW;
-alias COLOR_MAGENTA = COLOR.COLOR_MAGENTA;
-alias COLOR_CYAN = COLOR.COLOR_CYAN;
-alias COLOR_WHITE = COLOR.COLOR_WHITE;
-
-version (Windows)
+struct Console
 {
-    extern (C++) static WORD consoleAttributes(HANDLE h)
+    version (Windows)
     {
-        static __gshared CONSOLE_SCREEN_BUFFER_INFO sbi;
-        static __gshared bool sbi_inited = false;
-        if (!sbi_inited)
-            sbi_inited = GetConsoleScreenBufferInfo(h, &sbi) != FALSE;
-        return sbi.wAttributes;
-    }
+        import core.sys.windows.windows;
 
-    enum : int
-    {
-        FOREGROUND_WHITE = FOREGROUND_RED | FOREGROUND_GREEN | FOREGROUND_BLUE,
-    }
-}
+      private:
+        CONSOLE_SCREEN_BUFFER_INFO sbi;
+        HANDLE handle;
+        FILE* _fp;
 
-extern (C++) bool isConsoleColorSupported()
-{
-    version (CRuntime_DigitalMars)
-    {
-        return isatty(stderr._file) != 0;
-    }
-    else version (CRuntime_Microsoft)
-    {
-        return isatty(fileno(stderr)) != 0;
+      public:
+
+        @property FILE* fp() { return _fp; }
+
+        /*********************************
+         * Create an instance of Console connected to stream fp.
+         * Params:
+         *      fp = io stream
+         * Returns:
+         *      pointer to created Console
+         *      null if failed
+         */
+        static Console* create(FILE* fp)
+        {
+            /* Determine if stream fp is a console
+             */
+            version (CRuntime_DigitalMars)
+            {
+                if (!isatty(fp._file))
+                    return null;
+            }
+            else version (CRuntime_Microsoft)
+            {
+                if (!isatty(fileno(fp)))
+                    return null;
+            }
+            else
+            {
+                return null;
+            }
+
+            DWORD nStdHandle;
+            if (fp == stdout)
+                nStdHandle = STD_OUTPUT_HANDLE;
+            else if (fp == stderr)
+                nStdHandle = STD_ERROR_HANDLE;
+            else
+                return null;
+
+            auto h = GetStdHandle(nStdHandle);
+            CONSOLE_SCREEN_BUFFER_INFO sbi;
+            if (GetConsoleScreenBufferInfo(h, &sbi) == 0) // get initial state of console
+                return null;
+
+            auto c = new Console();
+            c._fp = fp;
+            c.handle = h;
+            c.sbi = sbi;
+            return c;
+        }
+
+        /*******************
+         * Turn on/off intensity.
+         * Params:
+         *      bright = turn it on
+         */
+        void setColorBright(bool bright)
```

Comment:
Should this be name `colorBright` instead to allow to be used as a property, i.e. `console.colorBright = false`?